### PR TITLE
Expire first party cookies

### PR DESCRIPTION
**Pull request by marcosholgado**
Task/Issue URL: https://app.asana.com/0/488551667048375/1203078418143031/f 

### Description
This PR changes the expiry date for 1st party cookies based on our remote config to avoid further tracking.

### Steps to test this PR

Note: In the task you will find a json file to be used with Charles or similar. Configure it and map https://staticcdn.duckduckgo.com/trackerblocking/config/v2/android-config.json to it.

_Remote config enabled_
- [x] Install the app.
- [x] Go to `https://privacy-test-pages.glitch.me/privacy-protections/storage-blocking/`.
- [x] Click on 'store data', this will store cookies in the db.
- [x] Take the app to the background
- [x] In the device file explorer go to `data/data/com.duckduckgo.mobile.android.debug/app_webview/Default`
- [x] Synchronize the Cookies file (the db) using right click to make sure you have the latest version
- [x] Download the file
- [x] Open the DB using something like `https://sqlitebrowser.org` or similar
- [x] Go to the cookies table
- [x] All the cookies that have `is_httponly = 0` should have an `expires_utc` value of 7 days from today.
- [x] To check the above, copy the `expires_utc` value and use `https://www.epochconverter.com/webkit` to convert to a readable format.
- [x] Take note of the value
- [x] Close and restart the app
- [x] Take the app to the background and download the Cookies DB again.
- [x] The `expires_utc` values should remain the same.
- [x] You should also now see a cookie with `is_httponly=1`. The value of that cookie should be greater than 7 days from today.
- [x] Continue with the test case

_User allow list_
- [x] Open the app 
- [x] Go to `https://privacy-test-pages.glitch.me/privacy-protections/storage-blocking/`.
- [x] Disable protections
- [x] Click on store data
- [x] Send the app to the background and download the Cookies DB.
- [x] All the values should be greater than 7 days from today.

_User allow list_
- [x] Change the unprotected temporary section in the json config and add `privacy-test-pages.glitch.me` as an unprotected domain.
- [x] Install the app
- [x] Open the app 
- [x] Go to `https://privacy-test-pages.glitch.me/privacy-protections/storage-blocking/`.
- [x] Click on store data
- [x] Send the app to the background and download the Cookies DB.
- [x] All the values should be greater than 7 days from today.

_Exceptions_
- [x] Undo the change in the unprotected temporary section and add `privacy-test-pages.glitch.me` in the exceptions list of the cookie feature.
- [x] Install the app
- [x] Open the app 
- [x] Go to `https://privacy-test-pages.glitch.me/privacy-protections/storage-blocking/`.
- [x] Click on store data
- [x] Send the app to the background and download the Cookies DB.
- [x] All the values should be greater than 7 days from today.

_Feature disabled_
- [x] Undo the change in the cookies section and change the cookie feature from enabled to disabled.
- [x] Install the app
- [x] Open the app 
- [x] Go to `https://privacy-test-pages.glitch.me/privacy-protections/storage-blocking/`.
- [x] Click on store data
- [x] Send the app to the background and download the Cookies DB.
- [x] All the values should be greater than 7 days from today.

### CODE REVIEW

**Comment by marcosholgado** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2362** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2362" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2362?utm_source=stack-comment).

File: privacy-config/privacy-config-impl/src/main/java/com/duckduckgo/privacy/config/impl/features/unprotectedtemporary/RealUnprotectedTemporary.kt [hunk 1]
```diff
@@ -28,6 +28,8 @@ import dagger.SingleInstanceIn
 @SingleInstanceIn(AppScope::class)
 class RealUnprotectedTemporary @Inject constructor(private val repository: UnprotectedTemporaryRepository) : UnprotectedTemporary {
 
+    override fun allExceptions(): List<String> = repository.exceptions.map { it.domain }
```
Comment (joshliebe) [inline privacy-config/privacy-config-impl/src/main/java/com/duckduckgo/privacy/config/impl/features/unprotectedtemporary/RealUnprotectedTemporary.kt:31]: Perhaps rename this to `allExceptionDomains`

File: privacy-config/privacy-config-api/src/main/java/com/duckduckgo/privacy/config/api/UnprotectedTemporary.kt [hunk 2]
```diff
@@ -25,4 +25,9 @@ interface UnprotectedTemporary {
      * otherwise.
      */
     fun isAnException(url: String): Boolean
+
+    /**
+     * This method returns a list of Strings containing the domains in the unprotected temporary exceptions list.
+     */
+    fun allExceptions(): List<String>
```
Comment (joshliebe) [inline privacy-config/privacy-config-api/src/main/java/com/duckduckgo/privacy/config/api/UnprotectedTemporary.kt:32]: Perhaps rename this to `allExceptionDomains`

File: cookies/cookies-impl/src/main/java/com/duckduckgo/cookies/impl/di/CookiesModule.kt [hunk 3]
```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2021 DuckDuckGo
```
Comment (joshliebe) [inline cookies/cookies-impl/src/main/java/com/duckduckgo/cookies/impl/di/CookiesModule.kt:2]:
```suggestion
 * Copyright (c) 2022 DuckDuckGo
```

File: cookies/cookies-impl/src/test/java/com/duckduckgo/cookies/impl/features/CookiesFeatureTogglesPluginTest.kt [hunk 4]
```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2021 DuckDuckGo
```
Comment (joshliebe) [inline cookies/cookies-impl/src/test/java/com/duckduckgo/cookies/impl/features/CookiesFeatureTogglesPluginTest.kt:2]:
```suggestion
 * Copyright (c) 2022 DuckDuckGo
```

Comment (marcosholgado) [reply cookies/cookies-impl/src/test/java/com/duckduckgo/cookies/impl/features/CookiesFeatureTogglesPluginTest.kt]: 👍 done!

File: cookies/cookies-impl/src/main/java/com/duckduckgo/cookies/impl/di/CookiesModule.kt (hunk 3 above)
Comment (marcosholgado) [reply cookies/cookies-impl/src/main/java/com/duckduckgo/cookies/impl/di/CookiesModule.kt]: 👍 done!

File: privacy-config/privacy-config-api/src/main/java/com/duckduckgo/privacy/config/api/UnprotectedTemporary.kt (hunk 2 above)
Comment (marcosholgado) [reply privacy-config/privacy-config-api/src/main/java/com/duckduckgo/privacy/config/api/UnprotectedTemporary.kt]: I've removed this API to use the one you created for content scripts instead :)

**Review by joshliebe — APPROVED** [review]
Tested all scenarios and works as expected, LGTM! 🚀

**Pull request merged**